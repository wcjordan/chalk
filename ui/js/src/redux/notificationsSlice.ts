import { createSlice } from '@reduxjs/toolkit';

import { NotificationsState } from './types';

const initialState: NotificationsState = {
  notificationQueue: [],
};
export default createSlice({
  name: 'notifications',
  initialState,
  reducers: {
    addNotification: (state, action) => {
      state.notificationQueue.push(action.payload);
    },
    dismissNotification: (state) => {
      state.notificationQueue.shift();
    },
  },
});
