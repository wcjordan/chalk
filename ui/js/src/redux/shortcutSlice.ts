import { createSlice } from '@reduxjs/toolkit';

import { ShortcutState } from './types';

const initialState: ShortcutState = {
  operations: [],
  latestGeneration: 0,
};
export default createSlice({
  name: 'shortcuts',
  initialState,
  reducers: {
    addEditTodoOperation: (state, action) => {
      state.operations.push({
        type: 'EDIT_TODO',
        payload: action.payload,
        generation: state.latestGeneration,
      });
    },
    addMoveTodoOperation: (state, action) => {
      state.operations.push({
        type: 'MOVE_TODO',
        payload: action.payload,
        generation: state.latestGeneration,
      });
    },
    incrementGenerations: (state) => {
      state.latestGeneration = state.latestGeneration + 1;
    },
    clearOperationsUpThroughGeneration: (state, action) => {
      state.operations = state.operations.filter(
        (op) => op.generation > action.payload,
      );
    },
  },
});
