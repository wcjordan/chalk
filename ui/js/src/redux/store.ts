import { configureStore } from '@reduxjs/toolkit';
import rootReducer, { listTodos } from './reducers';

const store = configureStore({
  reducer: rootReducer,
});

// TODO dispose
window.setInterval(() => store.dispatch(listTodos()), 1000);

export default store;
