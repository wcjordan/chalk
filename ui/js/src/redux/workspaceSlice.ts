import { Platform } from 'react-native';
import { createSlice } from '@reduxjs/toolkit';

import { FILTER_STATUS, WorkContext, WorkspaceState } from './types';

// TODO (jordan) Improve to support or'd label groups
// Will be useful for adding Chalk + vague to Chalk Planning
export const workContexts: { [key: string]: WorkContext } = {
  inbox: {
    displayName: 'Inbox',
    labels: {
      Unlabeled: FILTER_STATUS.Active,
    },
  },
  urgent: {
    displayName: 'Urgent',
    labels: {
      urgent: FILTER_STATUS.Active,
    },
  },
  chores: {
    displayName: 'Chores',
    labels: {
      errand: FILTER_STATUS.Active,
    },
  },
  quickFixes: {
    displayName: 'Quick Fixes',
    labels: {
      '5 minutes': FILTER_STATUS.Active,
    },
  },
  chalkCoding: {
    displayName: 'Chalk Coding',
    labels: {
      Chalk: FILTER_STATUS.Active,
      '25 minutes': FILTER_STATUS.Active,
    },
  },
  chalkPlanning: {
    displayName: 'Chalk Planning',
    labels: {
      Chalk: FILTER_STATUS.Active,
      '60 minutes': FILTER_STATUS.Active,
    },
  },
};

const initialState: WorkspaceState = {
  csrfToken: null,
  labelTodoId: null,
  filterLabels: {
    Unlabeled: FILTER_STATUS.Active,
  },
  loggedIn: Platform.OS === 'web',
  editTodoId: null,
};
export default createSlice({
  name: 'workspace',
  initialState,
  reducers: {
    filterByLabels: (state, action) => {
      state.filterLabels = Object.assign({}, action.payload);
    },
    logIn: (state, action) => {
      state.loggedIn = action.payload.loggedIn;
      state.csrfToken = action.payload.csrfToken;
    },
    setEditTodoId: (state, action) => {
      state.editTodoId = action.payload;
    },
    setLabelTodoId: (state, action) => {
      state.labelTodoId = action.payload;
    },
    setWorkContext: (state, action) => {
      const workContext = workContexts[action.payload];
      if (workContext) {
        state.filterLabels = Object.assign({}, workContext.labels);
      }
    },
  },
});
