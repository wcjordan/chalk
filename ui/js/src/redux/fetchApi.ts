import Cookies from 'js-cookie';
import { Platform } from 'react-native';

import { getEnvFlags } from '../helpers';
import { RootState } from './store';

const webCsrfToken = Cookies.get('csrftoken') as string;
export function getCsrfToken(getState: () => RootState): string {
  if (Platform.OS === 'web') {
    return webCsrfToken;
  }
  return getState().workspace.csrfToken as string;
}

export async function list<T>(apiUri: string): Promise<Array<T>> {
  const response = await fetch(apiUri, getRequestOpts('GET'));
  return handleResponse<Array<T>>(response);
}

export async function patch<T, Q extends { id: number }>(
  apiUri: string,
  entryPatch: Q,
  csrfToken: string,
): Promise<T> {
  const requestOpts = getRequestOpts('PATCH', csrfToken);
  requestOpts.body = JSON.stringify(entryPatch);
  const response = await fetch(`${apiUri}${entryPatch.id}/`, requestOpts);
  return handleResponse<T>(response);
}

export async function create<T, Q>(
  apiUri: string,
  newEntry: Q,
  csrfToken: string,
): Promise<T> {
  return postRequest<T, Q>(apiUri, newEntry, csrfToken);
}

export async function postRequest<T, Q>(
  uri: string,
  data: Q,
  csrfToken: string,
): Promise<T> {
  const requestOpts = getRequestOpts('POST', csrfToken);
  requestOpts.body = JSON.stringify(data);
  const response = await fetch(uri, requestOpts);
  return handleResponse<T>(response);
}

function getRequestOpts(
  method: string,
  csrfToken: string | null = null,
): RequestInit {
  const headers: HeadersInit = {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  };
  if (csrfToken) {
    headers['X-CSRFToken'] = csrfToken;
  }

  const credentials: RequestCredentials = Platform.select({
    native: 'include',
    default: 'same-origin',
  });

  return {
    credentials: credentials,
    headers: headers,
    method: method,
  };
}

async function handleResponse<T>(response: Response): Promise<T> {
  if (!response.ok) {
    throw new Error(`${response.status} ${response.statusText}`);
  }
  return await response.json();
}

export function getWsRoot(): string {
  if (Platform.OS === 'web') {
    return '';
  }

  const subdomain =
    getEnvFlags().ENVIRONMENT === 'prod' ? 'chalk' : 'chalk-dev';
  return `http://${subdomain}.flipperkid.com/`
}

// Used to exchange login token for session cookie in mobile login flow
export async function completeAuthCallback(token: string) {
  return await fetch(
    `${getWsRoot()}api/todos/auth_callback/?code=${token}`,
    getRequestOpts('GET'),
  );
}

export async function logSessionData(
  dataString: string,
  csrfToken: string,
): Promise<string> {
  
  const requestOpts = getRequestOpts('POST', csrfToken);
  requestOpts.body = dataString;
  const response = await fetch(
    `${getWsRoot()}api/todos/log_session_data/`,
    requestOpts,
  );
  return handleResponse<string>(response);
}
