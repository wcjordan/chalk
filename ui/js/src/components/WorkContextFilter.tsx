import React, { useCallback } from 'react';
import { StyleSheet, View, ViewStyle } from 'react-native';
import { FILTER_STATUS, WorkContext } from '../redux/types';
import FilterViewControls from './FilterViewControls';
import LabelChip from './LabelChip';

interface Style {
  filterView: ViewStyle;
  spacer: ViewStyle;
}

const styles = StyleSheet.create<Style>({
  filterView: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    width: '100%',
    backgroundColor: '#d9f0ffff',
  },
  spacer: {
    flexGrow: 1,
  },
});

const WorkContextFilter: React.FC<Props> = function (props: Props) {
  const {
    activeWorkContext,
    setWorkContext,
    showCompletedTodos,
    showLabelFilter,
    toggleShowCompletedTodos,
    toggleShowLabelFilter,
    workContexts,
  } = props;

  const setWorkContextCb = useCallback(
    (workContext: string) => {
      setWorkContext(workContext);
    },
    [setWorkContext],
  );

  const chips = Object.keys(workContexts).map((workContext) => (
    <LabelChip
      display={workContexts[workContext].displayName}
      key={workContext}
      label={workContext}
      onPress={setWorkContextCb}
      status={
        workContext === activeWorkContext ? FILTER_STATUS.Active : undefined
      }
    />
  ));
  return (
    <View style={styles.filterView} testID="work-context-filter">
      {chips}
      <View style={styles.spacer} />
      <FilterViewControls
        showCompletedTodos={showCompletedTodos}
        showLabelFilter={showLabelFilter}
        toggleShowCompletedTodos={toggleShowCompletedTodos}
        toggleShowLabelFilter={toggleShowLabelFilter}
      />
    </View>
  );
};

type Props = {
  activeWorkContext: string | undefined;
  setWorkContext: (workContext: string) => void;
  showCompletedTodos: boolean;
  showLabelFilter: boolean;
  toggleShowCompletedTodos: () => void;
  toggleShowLabelFilter: () => void;
  workContexts: { [key: string]: WorkContext };
};

export default WorkContextFilter;
