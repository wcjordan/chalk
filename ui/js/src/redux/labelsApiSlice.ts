import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { ApiState, ReduxState, Label } from './types';
import { getWsRoot, list } from './fetchApi';

const API_NAME = 'labelsApi';

// Exported for testing
export function getLabelsApi(): string {
  return `${getWsRoot()}api/todos/labels/`;
}

const initialState: ApiState<Label> = {
  entries: [],
  initialLoad: true,
  loading: false,
};

export const listLabels = createAsyncThunk<
  Label[],
  void,
  { state: ReduxState }
>(`${API_NAME}/list`, async () => list<Label>(getLabelsApi()), {
  condition: (_unused, { getState }) => {
    return !getState().labelsApi.loading;
  },
});

export default createSlice({
  name: API_NAME,
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(listLabels.pending, (state) => {
        state.loading = true;
      })
      .addCase(listLabels.fulfilled, (state, action) => {
        state.initialLoad = false;
        state.loading = false;
        state.entries = action.payload;
      })
      .addCase(listLabels.rejected, (state, action) => {
        state.initialLoad = false;
        state.loading = false;
        console.warn(`Loading Labels failed. ${action.error.message}`);
      });
  },
});
