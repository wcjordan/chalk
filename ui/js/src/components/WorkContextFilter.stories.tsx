import React from 'react';
import { workContexts } from '../redux/workspaceSlice';
import WorkContextFilter from './WorkContextFilter';

const defaultProps = {
  activeWorkContext: undefined,
  workContexts: workContexts,
};

export default {
  title: 'Work Context Filter',
  component: WorkContextFilter,
};
export const DefaultWorkContextFilter: React.FC = () => (
  <WorkContextFilter {...defaultProps} />
);

export const ActiveWorkContextFilter: React.FC = () => (
  <WorkContextFilter {...defaultProps} activeWorkContext="urgent" />
);
