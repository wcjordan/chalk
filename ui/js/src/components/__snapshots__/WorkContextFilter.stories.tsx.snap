// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work Context Filter ActiveWorkContextFilter smoke-test 1`] = `
<div class="css-175oi2r r-78d8zt r-18u37iz r-1w6e6rj r-13qz1uu"
     data-testid="work-context-filter"
>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Inbox
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-1iizze8 r-1bq2mok r-105ug2t"
       data-testid="chip-active-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip-active"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div class="css-175oi2r r-1kihuf0 r-tuq35u r-1m04atk r-9aemit">
          <div dir="auto"
               role="img"
               tabindex="-1"
               class="css-146c3p1 r-1niwhzg r-lrvibr r-633pao"
               style="font-size: 18px; color: rgb(29, 25, 43); transform: scaleX(1); line-height: 18px; font-family: material-community; font-weight: normal; font-style: normal;"
          >
            󰄬
          </div>
        </div>
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Urgent
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Quick Fixes
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Up Next
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Work
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Shopping
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Chalk Planning
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-16y2uox">
  </div>
  <div dir="auto"
       class="css-146c3p1 r-1pyaxff"
  >
    <div class="css-175oi2r r-xoduu5 r-1udh08x r-crgep1 r-105ug2t"
         data-testid="show-labels-container"
         style="background-color: rgba(0, 0, 0, 0); width: 36px; height: 36px; border-width: 0px; border-radius: 18px; border-color: rgb(121, 116, 126);"
    >
      <button role="button"
              tabindex="0"
              class="css-175oi2r r-xoduu5 r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-1awozwy r-16y2uox r-1777fci"
              data-testid="show-labels"
              type="button"
              style="border-radius: 18px;"
      >
        <span role="img"
              tabindex="-1"
              class="css-1jxf684 r-1niwhzg r-lrvibr r-633pao"
              style="font-size: 20px; color: rgb(0, 0, 0); transform: scaleX(1); line-height: 20px; font-family: material-community; font-weight: normal; font-style: normal;"
        >
          󱋷
        </span>
      </button>
    </div>
    <div class="css-175oi2r r-xoduu5 r-1udh08x r-crgep1 r-105ug2t"
         data-testid="show-completed-todos-container"
         style="background-color: rgba(0, 0, 0, 0); width: 36px; height: 36px; border-width: 0px; border-radius: 18px; border-color: rgb(121, 116, 126);"
    >
      <button role="button"
              tabindex="0"
              class="css-175oi2r r-xoduu5 r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-1awozwy r-16y2uox r-1777fci"
              data-testid="show-completed-todos"
              type="button"
              style="border-radius: 18px;"
      >
        <span role="img"
              tabindex="-1"
              class="css-1jxf684 r-1niwhzg r-lrvibr r-633pao"
              style="font-size: 20px; color: rgb(0, 0, 0); transform: scaleX(1); line-height: 20px; font-family: material-community; font-weight: normal; font-style: normal;"
        >
          󰄱
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Work Context Filter DefaultWorkContextFilter smoke-test 1`] = `
<div class="css-175oi2r r-78d8zt r-18u37iz r-1w6e6rj r-13qz1uu"
     data-testid="work-context-filter"
>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Inbox
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Urgent
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Quick Fixes
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Up Next
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Work
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Shopping
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-1phboty r-18u37iz r-1yadl64 r-78d8zt r-1bq2mok r-105ug2t"
       data-testid="chip-container"
       style="border-color: rgb(121, 116, 126); border-radius: 8px; box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px;"
  >
    <button role="button"
            tabindex="0"
            class="css-175oi2r r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-13qz1uu"
            data-testid="chip"
            type="button"
            style="border-radius: 8px;"
    >
      <div class="css-175oi2r r-1awozwy r-18u37iz r-bnwqim r-gy4na3"
           style="padding-right: 0px;"
      >
        <div dir="auto"
             class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a r-fdjqy7 r-ws14 r-9iso6 r-1x2decy r-15zivkp r-1jkjb r-1kb76zh r-14gqq1x r-lrvibr"
             style="direction: ltr; color: rgb(29, 25, 43); letter-spacing: 0.1px; font-weight: 500; line-height: 20px; font-size: 14px;"
        >
          Chalk Planning
        </div>
      </div>
    </button>
  </div>
  <div class="css-175oi2r r-16y2uox">
  </div>
  <div dir="auto"
       class="css-146c3p1 r-1pyaxff"
  >
    <div class="css-175oi2r r-xoduu5 r-1udh08x r-crgep1 r-105ug2t"
         data-testid="show-labels-container"
         style="background-color: rgba(0, 0, 0, 0); width: 36px; height: 36px; border-width: 0px; border-radius: 18px; border-color: rgb(121, 116, 126);"
    >
      <button role="button"
              tabindex="0"
              class="css-175oi2r r-xoduu5 r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-1awozwy r-16y2uox r-1777fci"
              data-testid="show-labels"
              type="button"
              style="border-radius: 18px;"
      >
        <span role="img"
              tabindex="-1"
              class="css-1jxf684 r-1niwhzg r-lrvibr r-633pao"
              style="font-size: 20px; color: rgb(0, 0, 0); transform: scaleX(1); line-height: 20px; font-family: material-community; font-weight: normal; font-style: normal;"
        >
          󱋷
        </span>
      </button>
    </div>
    <div class="css-175oi2r r-xoduu5 r-1udh08x r-crgep1 r-105ug2t"
         data-testid="show-completed-todos-container"
         style="background-color: rgba(0, 0, 0, 0); width: 36px; height: 36px; border-width: 0px; border-radius: 18px; border-color: rgb(121, 116, 126);"
    >
      <button role="button"
              tabindex="0"
              class="css-175oi2r r-xoduu5 r-1otgn73 r-1loqt21 r-bnwqim r-ctqt5z r-1udh08x r-1awozwy r-16y2uox r-1777fci"
              data-testid="show-completed-todos"
              type="button"
              style="border-radius: 18px;"
      >
        <span role="img"
              tabindex="-1"
              class="css-1jxf684 r-1niwhzg r-lrvibr r-633pao"
              style="font-size: 20px; color: rgb(0, 0, 0); transform: scaleX(1); line-height: 20px; font-family: material-community; font-weight: normal; font-style: normal;"
        >
          󰄱
        </span>
      </button>
    </div>
  </div>
</div>
`;
